### PR TITLE
Getting user info from backend to the settings page.

### PR DESCRIPTION
**Pull request by adityakushwah112**

### CODE REVIEW

File: app/src/main/java/com/del/delcontainer/ui/settings/SettingsFragment.java [hunk 1]
```diff
@@ -85,4 +96,12 @@ public void onChanged(@Nullable String s) {
 
         return root;
     }
+
+    /**
+     * Calls the del-api service to get the first name linked to the current user
+     */
+    private void getFirstName(){
+        //settingsViewModel.getUserFirstName(authRepository.getAccessToken(), authRepository.getUserId());
```
Comment (benphilip1991) [inline app/src/main/java/com/del/delcontainer/ui/settings/SettingsFragment.java:104]: Get rid of commented out code, please!

File: app/src/main/java/com/del/delcontainer/ui/settings/SettingsViewModel.java [hunk 2]
```diff
@@ -1,19 +1,61 @@
...
+public class SettingsViewModel extends ViewModel {
 
-    public SettingsViewModel() {
-        mText = new MutableLiveData<>();
-        mText.setValue("User Name");
+    private static final String TAG = "SettingsViewModel";
+    private MutableLiveData<String> firstName = new MutableLiveData<>();
+    private MutableLiveData<LoginStateRepo> loginStateRepo = new MutableLiveData<>();
+    public MutableLiveData<LoginStateRepo> getLoginStateRepo() {
+        return loginStateRepo;
```
Comment (benphilip1991) [inline app/src/main/java/com/del/delcontainer/ui/settings/SettingsViewModel.java:26]: What are you trying to do here?

File: app/src/main/java/com/del/delcontainer/ui/settings/SettingsViewModel.java [hunk 3]
```diff
@@ -1,19 +1,61 @@
...
-        return mText;
+    Retrofit retrofit = APIUtils.getApiClient();
+
+    /**
+     * Fetch first name of the user
+     *
+     * @param token
+     * @param userId
+     */
+    public void getUserFirstName(String token, String userId){
```
Comment (benphilip1991) [inline app/src/main/java/com/del/delcontainer/ui/settings/SettingsViewModel.java:37]: Why just the first name? Are there any other details that you can fetch?

File: app/src/main/java/com/del/delcontainer/ui/settings/SettingsFragment.java (hunk 1 above)
Comment (adityakushwah112) [reply app/src/main/java/com/del/delcontainer/ui/settings/SettingsFragment.java:104]: I kept it for testing but forgot to remove it. I’ll delete those.

File: app/src/main/java/com/del/delcontainer/ui/settings/SettingsViewModel.java (hunk 2 above)
Comment (adityakushwah112) [reply app/src/main/java/com/del/delcontainer/ui/settings/SettingsViewModel.java:26]: Sorry, I forgot to remove the 24 and 25. They’re not doing anything. I’ll delete them as well.

File: app/src/main/java/com/del/delcontainer/ui/settings/SettingsViewModel.java (hunk 3 above)
Comment (adityakushwah112) [reply app/src/main/java/com/del/delcontainer/ui/settings/SettingsViewModel.java:37]: Yes, only first name for now. I can similarly fetch other details, but I didn’t know exactly where to show them. Showcasing them in a separate fragment or dialogue would take more hours of setting up UI so I couldn’t find doing that after Wednesday shift. Will start soon.

**Pull request closed without merging**